File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Expressions/Binary/BinaryExpression.cs
using System;
using UvA.SoftCon.Questionnaire.Common;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions.Binary
{
    public abstract class BinaryExpression : Expression
    {
        public Operation Operation
        {
            get;
            private set;
        }

        public Expression Left
        {
            get;
            private set;
        }

        public Expression Right
        {
            get;
            private set;
        }

        protected BinaryExpression(Operation operation, Expression left, Expression right, TextPosition position)
            : base(position)
        {
            Operation = operation;
            Left = left;
            Right = right;
        }

        public abstract bool OperandTypesAreValid(DataType leftType, DataType rightType);

        public override string ToString()
        {
            return String.Format("{0} {1} {2}", Left.ToString(), StringEnum.GetStringValue(Operation), Right.ToString());
        }
    }
}

File: nlamah/src/main/java/org/nlamah/QL/Binary/BinaryExpression.java
package org.nlamah.QL.Binary;

import org.nlamah.QL.Expression.Expression;

public abstract class BinaryExpression extends Expression 
{
	private Expression leftHandExpression;
	private Expression rightHandExpression;
	
	public BinaryExpression(Expression leftHandExpression, Expression rightHandExpression)
	{
		this.leftHandExpression = leftHandExpression;
		this.rightHandExpression = rightHandExpression;
	}
	
	public Expression leftHandExpression()
	{
		return this.leftHandExpression;
	}
	
	public Expression rightHandExpression()
	{
		return this.rightHandExpression;
	}
}
